service/common/src/main/java/org/apache/polaris/service/catalog/io/WasbTranslatingFileIOFactory.java

```diff
@@ -19,19 +19,63 @@
 package org.apache.polaris.service.catalog.io;
 
 import io.smallrye.common.annotation.Identifier;
+import jakarta.annotation.Nonnull;
 import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
 import java.util.Map;
-import org.apache.hadoop.conf.Configuration;
-import org.apache.iceberg.CatalogUtil;
+import java.util.Set;
+import org.apache.iceberg.catalog.TableIdentifier;
 import org.apache.iceberg.io.FileIO;
+import org.apache.polaris.core.PolarisConfigurationStore;
+import org.apache.polaris.core.context.RealmId;
+import org.apache.polaris.core.persistence.MetaStoreManagerFactory;
+import org.apache.polaris.core.persistence.PolarisResolvedPathWrapper;
+import org.apache.polaris.core.storage.PolarisStorageActions;
+import org.apache.polaris.service.config.RealmEntityManagerFactory;
 
 /** A {@link FileIOFactory} that translates WASB paths to ABFS ones */
 @ApplicationScoped
 @Identifier("wasb")
 public class WasbTranslatingFileIOFactory implements FileIOFactory {
+
+  private final FileIOFactory defaultFileIOFactory;
+
+  @Inject
+  public WasbTranslatingFileIOFactory(
+      RealmEntityManagerFactory realmEntityManagerFactory,
+      MetaStoreManagerFactory metaStoreManagerFactory,
+      PolarisConfigurationStore configurationStore) {
+    defaultFileIOFactory =
+        new DefaultFileIOFactory(
```

Comment:
How is the behaviour of `WasbTranslatingFileIOFactory` different from `DefaultFileIOFactory`? Why do we need `WasbTranslatingFileIOFactory`?

Reply:
When we load the FileIO, we wrap the FileIO loaded by `DefaultFileIOFactory` with `WasbTranslatingFileIO`. `WasbTranslatingFileIO` will traslate the `wasb` file path to `abfs` file path.

Reply:
I guess I missed `WasbTranslatingFileIO` :sweat_smile:  IMHO, wrapping an implementation that has a specific purpose and behaviour (like `DefaultFileIOFactory`) is not ideal. I'd prefer to extract a base class that can be shared between `DefaultFileIOFactory` and `WasbTranslatingFileIOFactory`... but that can be done later.

Reply:
Will do it in a separate PR! Another dumb question, is it the best practice to extend a abstract for an ApplicationScoped bean considering this? https://github.com/apache/polaris/pull/724#discussion_r1925795624

Reply:
@dimas-b  we didn't take that approach because `WasbTranslatingFileIO` was/is supposed to be very temporary until the Iceberg FileIO implementations support WASB. This was originally merged, then reverted, so I'm not sure on the current status. But the goal should remain to make `WasbTranslatingFileIO` easy to rip out.

Reply:
@XJDKC : I believe @snazy 's concern was about extending classes that are themselves CDI beans (i.e. have CDI annotations). I do not think there's any problem with extending simple java classes and putting CDI annotations on the child class.

Reply:
@eric-maynard : like I commented we can address this later... If we can remove `WasbTranslatingFileIO` at that point that will remove my concern too :)

Reply:
I see, maybe we can keep this as it's since the long term plan is to remove it.